[ADD] Referrals: new rewards doc (#9615)
odoo/documentation: content/applications/hr/referrals.rst

all of which must be installed in order for the *Referrals* application to function.

> The only configurations needed for the *Referrals* application *after* installation are related to

Reviewer: ```suggestion
The only configurations needed for the *Referrals* application *after* it has been installed, are related to
```
SUGGESTION to clean up the wording/pacing of the sentence a bit

> the :doc:`rewards <referrals/rewards>` ; everything else is pre-configured when Odoo *Referrals*

Reviewer: ```suggestion
the :doc:`rewards <referrals/rewards>`; everything else is pre-configured when Odoo *Referrals*
```
no need for a space separating the end of the doc link and the semi-colon 👍

.. seealso::

>    :doc:`referrals/share_jobs`
>    :doc:`referrals/rewards`

Reviewer 2: Add bullets for multiple seealso items :)
```suggestion
   - :doc:`referrals/share_jobs`
   - :doc:`referrals/rewards`
```

Author: OMG I was used to removing the bullets when there was only one doc, I totally forgot!
